requirements.txt

```
@@ -11,7 +11,8 @@ imageio>=2.31.4
 joblib~=1.0
 matplotlib>=3.3.0
 nibabel~=3.2
-onnxruntime~=1.7
+onnx
+onnxruntime>=1.7.0,<1.16.0
```

A nitpick but I'd prefer not to have an upper bound 🤷 
```suggestion
onnxruntime~=1.7
```

---

Thank you so much for pointing this out!! 

I had set this upper bound because, at the time, `onnxruntime==1.16.0` contained a bug that was encountered on SCT's end: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4225

This bug has since been resolved, so it's sufficient to just set `onnxruntime>=1.7.0,!=1.16.0`

---

Just out of curiosity, why has the `~=` syntax been proposed? 

From what I understand, this syntax will limit `onnxruntime` to only e.g. `1.7.0`, `1.7.1`, `1.7.2`, etc. which limits compatibility with other packages.

---

Oh!! Thank you so much. It seems I've misunderstood what `~=` refers to.

In other words, since you've suggested `~=1.7`, this will include `1.8`, `1.9`, etc. but _not_ `2.*`, `3.*`, etc.

And, the behavior I describe above would only be true if `~=1.7.0` was specified instead.

In that case, I fully support your suggestion over `>=`. :)